src/reducers/entities/posts.js

```diff
@@ -359,6 +359,33 @@ function handleRemovePost(posts = {}, postsInChannel = {}, postsInThread = {}, a
     return {posts: nextPosts, postsInChannel: nextPostsForChannel, postsInThread: nextPostsForThread || postsInThread};
 }
 
+function handleRethreadPost(postsMap, action) {
+    const posts = postsMap.posts ? postsMap.posts : {};
+    const postsInChannel = postsMap.postsInThread ? postsMap.postsInChannel : {};
+    const postsInThread = postsMap.postsInThread ? postsMap.postsInThread : {};
+    var post = action.data.post;
+    const nextPosts = posts;
+    nextPosts[post.id] = post;
+    const rootId = action.data.original_root_id;
+
+    const nextPostsForChannel = postsInChannel;
```

Comment:
Is something supposed to be changing with `nextPostsInChannel`? The code that affects it just changes its name repeatedly
```
function handleRethreadPost(postsMap, action) {
    const postsInChannel = postsMap.postsInThread ? postsMap.postsInChannel : {};
    ...
    const nextPostsForChannel = postsInChannel;
    ...
    return {..., postsInChannel: nextPostsForChannel, ...};
}
```